crates/rlbot/src/scripts.rs

```diff
@@ -0,0 +1,90 @@
+use rlbot_flat::flat::{
+    BallPrediction, ConnectionSettings, FieldInfo, GamePacket, MatchComm, MatchConfiguration,
+};
+
+use crate::{
+    Packet, RLBotConnection, StartingInfo,
+    util::{PacketQueue, write_multiple_packets},
+};
+
+#[allow(unused_variables)]
+pub trait Script {
+    fn new(
+        agent_id: String,
+        match_config: MatchConfiguration,
+        field_info: FieldInfo,
+        packet_queue: &mut PacketQueue,
+    ) -> Self;
+    fn on_packet(&mut self, game_packet: GamePacket, packet_queue: &mut PacketQueue);
```

Comment:
either this should be called `tick` or we should rename all of them to `on_game_packet` imo

Reply:
There is no reason to enforce that imo, except for maybe keeping the percentages rendered in-game by core nice. I prefer the tick/on_packet function being consistent with the other on_x functions over forcing input though.